cmd/rekor-cli/app/timestamp.go

```diff
@@ -0,0 +1,198 @@
+	return nil
+}
+
+func validateTimestampFlags() error {
+	requestStr := viper.GetString("request")
+	fileStr := viper.GetString("file")
+	if requestStr == "" && fileStr == "" {
+		return errors.New("request or file must be specified")
+	}
+	return nil
+}
+
+func createRequestFromFlags() (*pkcs9.TimeStampReq, error) {
+	requestStr := viper.GetString("request")
+	var timestampReq *pkcs9.TimeStampReq
+	if requestStr == "" {
+		fileStr := viper.GetString("file")
+		fileBytes, err := ioutil.ReadFile(filepath.Clean(fileStr))
+		if err != nil {
+			return nil, fmt.Errorf("error reading request from file: %w", err)
+		}
+		timestampReq, err = util.TimestampRequestFromData(fileBytes)
+		if err != nil {
+			return nil, fmt.Errorf("error creating timestamp request: %w", err)
+		}
+	} else {
+		rawBytes, err := ioutil.ReadFile(filepath.Clean(requestStr))
+		if err != nil {
+			return nil, fmt.Errorf("error reading request from file: %w", err)
+		}
+		timestampReq, err = util.ParseTimestampRequest(rawBytes)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing timestamp request: %w", err)
+		}
+	}
+	return timestampReq, nil
+}
+
+type timestampCmdOutput struct {
+	Location string
+}
+
+func (t *timestampCmdOutput) String() string {
+	return fmt.Sprintf(`
+Wrote response to: %v
+`, t.Location)
+}
+
+var timestampCmd = &cobra.Command{
+	Use:   "timestamp",
+	Short: "Rekor timestamp command",
+	Long:  "Generates and uploads (WIP) an RFC 3161 timestamp response to the log. The timestamp response can be verified locally using Rekor's timestamping cert chain.",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if err := viper.BindPFlags(cmd.Flags()); err != nil {
+			log.Logger.Fatal("Error initializing cmd line args: ", err)
+		}
+		if err := validateTimestampFlags(); err != nil {
+			log.Logger.Error(err)
+			_ = cmd.Help()
+			return err
+		}
+		return nil
+	},
+	Run: format.WrapCmd(func(args []string) (interface{}, error) {
+		rekorClient, err := GetRekorClient(viper.GetString("rekor_server"))
+		if err != nil {
+			return nil, err
+		}
+
+		timestampReq, err := createRequestFromFlags()
+		if err != nil {
+			return nil, err
+		}
+		requestBytes, err := asn1.Marshal(*timestampReq)
+		if err != nil {
+			return nil, err
+		}
+
+		params := timestamp.NewGetTimestampResponseParams()
+		params.Query = &models.TimestampRequest{}
+		params.Query.RfcRequest = strfmt.Base64(requestBytes)
+
+		resp, err := rekorClient.Timestamp.GetTimestampResponse(params)
+		if err != nil {
+			return nil, err
+		}
+		body, err := base64.StdEncoding.DecodeString(resp.Payload.RfcResponse.String())
+		if err != nil {
+			return nil, err
+		}
+
+		// Sanity check response and check if the TimeStampToken was successfully created
+		_, err = timestampReq.ParseResponse(body)
+		if err != nil {
+			return nil, err
+		}
+
+		// Write response to file
+		outStr := viper.GetString("out")
+		if outStr == "" {
```

Comment:
Should we default to stdout maybe?

Reply:
Hmm it's kinda gross binary file. But tbh most of the time you pipe into a response.tsr after making a request, so yeah, why not.

Reply:
+1

Reply:
Ah, fair. Is there a standard pem encoding? If not then yeah keep it as the file I guess.

Reply:
Right I forgot the rationale here was that if I wrote to file, I could also include info on what the log index is when uploaded.